src/metabase/models/collection.clj

```diff
@@ -943,9 +943,12 @@
 (defn- parent-identity-hash [coll]
   (let [parent-id (-> coll
                       (t2/hydrate :parent_id)
-                      :parent_id)]
-    (if parent-id
-      (serdes/identity-hash (t2/select-one Collection :id parent-id))
+                      :parent_id)
+        parent (t2/select-one Collection :id parent-id)]
```

Comment:
It bothers me that we still perform a query when parent-id is nil 😞

Reply:
oh right, I'm weird 🙄